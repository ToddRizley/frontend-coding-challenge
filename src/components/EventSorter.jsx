import React from 'react';

const EventSorter = props => (
  <div className="event-sorter col-xs-6">
    <label className="sorter-label">Search Events By Title</label>
    <select className="col-xs-6"onChange={props.handleChange.bind(this)}>
      <option value="startTime">Sort By Start Time</option>
      <option value="title">Sort By Name</option>
    </select>
  </div>
);


EventSorter.propTypes = {
  handleChange: React.PropTypes.func,
};

export default EventSorter;
