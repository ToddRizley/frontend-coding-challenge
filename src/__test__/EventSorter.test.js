import React from 'react';
import EventSorter from '../components/EventSorter'
import renderer from 'react-test-renderer';

describe('<EventSorter/>', () => {
  var handleChange = jest.fn()
   it('renders correctly', () => {
    const rendered = renderer.create(
      <EventSorter handleChange={handleChange} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
