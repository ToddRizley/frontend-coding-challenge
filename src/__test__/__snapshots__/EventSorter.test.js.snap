// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventSorter/> renders correctly 1`] = `
<div
  className="event-sorter col-xs-6"
>
  <label
    className="sorter-label"
  >
    Search Events By Title
  </label>
  <select
    className="col-xs-6"
    onChange={[Function]}
  >
    <option
      value="startTime"
    >
      Sort By Start Time
    </option>
    <option
      value="title"
    >
      Sort By Name
    </option>
  </select>
</div>
`;
